### ISSUE TITLE
'function_exists' Returning False

### ISSUE THREAD

**Issue by Fenikkusu** [zephir-lang/zephir#1547]
Recently submitted a new PR to the cphalcon repo (https://github.com/phalcon/cphalcon/pull/12841). 

When I first started work, I used  `if function_exists(functionName)`. When running the unit tests using **`substr`**, I kept getting FALSE.

When I checked the raw C file, I saw that it was calling **`zephir_function_exists`**. I'm assuming that **`zephir_function_exists`** is incorrectly detecting the PHP internal **`substr`** function. 

I was able to get around this by using `call_user_func("function_exists", "substr")` instead, but it seems odd that the PHP internal would still return false.

For clarification: The PR for phalcon adds the ability to auto-register an array of existing functions, throwing an exception in the case the function doesn't exist. This could include both custom defined functions or PHP internal functions. When the option is set to TRUE, it defaults to all PHP internal functions.

**Comment by Jurigag** [comment]
Any script to reproduce? I just simply did:

```
var_dump(function_exists("substr"));
```
In zephir

And it correctly returns true on php 5.6,7.0,7.1

**Comment by Fenikkusu** [comment]
This occurred for https://github.com/twistersfury/cphalcon/tree/3.2.x-php-functions. Patch Below.

```diff
Index: phalcon/mvc/view/engine/volt/compiler.zep
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- phalcon/mvc/view/engine/volt/compiler.zep	(date 1495755658000)
+++ phalcon/mvc/view/engine/volt/compiler.zep	(revision )
@@ -2684,7 +2684,7 @@
 
 	    for functionName in configOptions {
 	        if !this->hasFunction(functionName) {
-	            if !call_user_func("function_exists", functionName) {
+	            if !function_exists(functionName) {
 	                throw new Exception(functionName . " does not exist");
 	            }
 

```

After compiling phalcon (`./vendor/bin/zephir build --buildEngine=ZendEngine3
`), running `./vendor/bin/codecept run unit Mvc/View/Engine/Volt/CompilerTest.php` will result in 
```
There were 2 errors:

---------
1) CompilerTest: Volt php functions | php native functions should work
 Test  tests/unit/Mvc/View/Engine/Volt/CompilerTest.php:testVoltPhpFunctions | PHP Native Functions Should Work
                                                                        
  [Phalcon\Mvc\View\Engine\Volt\Exception] zend_version does not exist  
                                                                        
#1  Phalcon\Mvc\View\Engine\Volt\Compiler->addPhpFunctions
#2  Phalcon\Mvc\View\Engine\Volt\Compiler->_compileSource
#3  /Users/fenikkusu/Documents/workspace/cphalcon/tests/unit/Mvc/View/Engine/Volt/CompilerTest.php:1275
#4  Phalcon\Test\Unit\Mvc\View\Engine\Volt\CompilerTest->Phalcon\Test\Unit\Mvc\View\Engine\Volt\{closure}
#5  /Users/fenikkusu/Documents/workspace/cphalcon/tests/unit/Mvc/View/Engine/Volt/CompilerTest.php:1279
#6  Phalcon\Test\Unit\Mvc\View\Engine\Volt\CompilerTest->testVoltPhpFunctions

---------
2) CompilerTest: Volt php functions array | php include array of functions should work
 Test  tests/unit/Mvc/View/Engine/Volt/CompilerTest.php:testVoltPhpFunctionsArray | PHP Include Array Of Functions Should Work
                                                                  
  [Phalcon\Mvc\View\Engine\Volt\Exception] substr does not exist  
                                                                  
#1  Phalcon\Mvc\View\Engine\Volt\Compiler->addPhpFunctions
#2  Phalcon\Mvc\View\Engine\Volt\Compiler->_compileSource
#3  /Users/fenikkusu/Documents/workspace/cphalcon/tests/unit/Mvc/View/Engine/Volt/CompilerTest.php:1294
#4  Phalcon\Test\Unit\Mvc\View\Engine\Volt\CompilerTest->Phalcon\Test\Unit\Mvc\View\Engine\Volt\{closure}
#5  /Users/fenikkusu/Documents/workspace/cphalcon/tests/unit/Mvc/View/Engine/Volt/CompilerTest.php:1298
#6  Phalcon\Test\Unit\Mvc\View\Engine\Volt\CompilerTest->testVoltPhpFunctionsArray

FAILURES!
Tests: 18, Assertions: 382, Errors: 2.

```

Running the code without the given patch runs correctly without errors. One thing I likely should note, I'm running this using ZendEngine3. I've not tried without.

**Comment by Jurigag** [comment]
Post code to reproduce this without phalcon dependency.

**Comment by sergeyklay** [comment]
The issue is:
```zep
var func = "substr";

function_exists("substr"); // TRUE
function_exists(func); // FALSE
```

This issue affected Zend Engine 3 only.

**Comment by sergeyklay** [comment]
Fixed in the version [**`0.9.10`**](https://github.com/phalcon/zephir/releases/tag/0.9.10). Thank you for contributing.
